src/PriorityQueue.php

```diff
@@ -63,13 +67,12 @@ class PriorityQueue implements Countable, IteratorAggregate, Serializable
      *
      * Priority defaults to 1 (low priority) if none provided.
      *
-     * @param  mixed $data
-     * @param  int $priority
-     * @return PriorityQueue
+     * @param  T         $data
+     * @param  TPriority $priority
+     * @return $this
      */
     public function insert($data, $priority = 1)
     {
-        $priority      = (int) $priority;
```

Comment:
I think we may need to keep this one, for BC? 🤔

Non-blocker though: I'm OK with consumers having to respect this API definition

Reply:
Psalm complains that `int` isn't a `TPriority` here - put it back in and suppress instead?

Reply:
Yes, suppress is fine